### ISSUE TITLE
allow file extension mapping to be extensible when mapping to kinds

### ISSUE THREAD

**Issue by rabbah** [apache/incubator-openwhisk-cli#286]
the CLI hardcodes some file extensions which it maps to action kinds - this list should be extensible, in two ways: 1) allow a build to incorporate custom mappings, 2) allow wksprops to encode additional preferences.

**Comment by csantanapr** [comment]
Yes this would make it easy for me to add `.sh` `--kind bash:4` and `.pl` `--kind perl:5` (openwhisk/action-docker-v2) the new go loop prooxy :-)

**Comment by mdeuser** [comment]
should this be driven by the backend since it knows exactly what runtimes it supports?

**Comment by rabbah** [comment]
i'm not entirely convinced myself that it should be a backend function although it would serve to normalize some of the extensions --- a client would need to fetch the runtime mapping and store them locally (it would be silly to query the backend on every create/update operation for a mapping which should be fairly static). Also some clients infer the kind without an extension (the ibm cloud ui for example).
